### ISSUE TITLE
Parameters are ignored in Rails 5.1

### ISSUE THREAD

**Issue by fernandomm** [activeadmin/inherited_resources#439]
I'm upgrading an old Rails 3.2 application to 5.1. This application uses inherited resources and I noticed that the following doesn't work:

```
collection_url(params.permit!.merge(example: true))
```

while this one works:

```
users_url(params.permit!.merge(example: true))
```

The issue seems to come from extract_options! in https://github.com/activeadmin/inherited_resources/blob/5d18499877852437f2c058747f07bc80e46557fc/lib/inherited_resources/url_helpers.rb#L223 and https://github.com/activeadmin/inherited_resources/blob/5d18499877852437f2c058747f07bc80e46557fc/lib/inherited_resources/url_helpers.rb#L229

Previously something like params.permit!.merge(example: true) would return a Hash and now it returns an ActionController::Parameters, which won't work with extract_options!.

I was able to fix this by using something like:

```
          def #{prefix}#{name}_url(*given_args)
            given_options = given_args.collect { |arg| arg.is_a?(ActionController::Parameters) ? arg.to_h : arg }.extract_options!
            #{prefix}#{segments}_url(#{ivars})
          end
```

If this change is correct, I can prepare a PR.

**Comment by rafaelfranca** [comment]
In my opinion it is your application responsibility to pass a Hash to that method. So it makes sense to you change your code to: `collection_url(params.permit!.to_h.merge(example: true))`

**Comment by fernandomm** [comment]
Yes, that's an option. But I guess that a lot of people expects the *_url and *_path methods that are defined by inherited resources to behave like the ones that are defined by Rails.

Also, considering that it worked fine in Rails < 5 and now it's silently failing, it looks like a bug and should be fixed. Or at least the type of the parameter should be checked and the developer should see some type of error.

I spent some time debugging until I figured out what was happening. Others may experience this problem too.
